index.js

```diff
@@ -0,0 +1,45 @@
+var express=require("express");
+var app=express();
+var arr = require('./halls.js');
+var duljina=arr.length;
+
+app.use(express.static('public'));
+app.get("/",function(req,res){
+    res.send("HELLO");
+});
+
+app.use("/halls",function(req,res){
+    res.json(arr);
+});
+app.use(express.urlencoded());
+app.post("/hallsCreate",function (req,res){
+    const formName = req.body.hallName;
+    arr[arr.length]={id:arr.length+1,name:formName,reservation:{isReserved:false,reservedFrom:null,reservedUntil:null}};
+    res.redirect("/halls");
+});
+app.post("/hallsUpdate",function (req,res){
+    var reservation=false;
+    var date=new Date();
+    if(req.body.isReserved=="isReserved"){
+        reservation=true;
+    }
+    for(var i=0;i<req.body.ID;i++){
```

Comment:
Na ovaj način nepotrebno prolaziš kroz objekte sve dok ne nađeš taj traženi objekt. Ako izostaviš for petlju i if provjeru, dobiješ ono što si trebao